lib/hls/hls_parser.js

```diff
@@ -985,14 +990,56 @@ shaka.hls.HlsParser = class {
       goog.asserts.assert(streamInfos.audio.length ||
           streamInfos.video.length, 'We should have created a stream!');
 
+      const drmInfos = [];
+      const keyIds = new Set();
+
+      if (sessionKeyTags.length > 0) {
+        const characteristics = tag.getAttributeValue('CHARACTERISTICS');
+        for (const drmTag of sessionKeyTags) {
+          const drmCharacteristics =
+              drmTag.getAttributeValue('CHARACTERISTICS');
+          if ((characteristics && !drmCharacteristics) ||
+              (!characteristics && drmCharacteristics) ||
+              (drmCharacteristics && characteristics &&
+              !characteristics.includes(drmCharacteristics))) {
+            continue;
+          }
+          const method = drmTag.getRequiredAttrValue('METHOD');
```

Comment:
This mostly looks like the same code that is being factored out into a method in PR #4568.  Would this be cleaner if we merged that PR first?